File: Kirameku-backend/admin/src/api/visitor.ts
import { http } from "@/utils/http";

export type VisitorItem = {
  id: number;
  ip: string;
  path: string;
  city: string;
  region: string;
  country: string;
  district: string;
  org: string;
  org_cn: string;
  asn: string;
  is_mobile: boolean;
  is_proxy: boolean;
  is_hosting: boolean;
  browser: string;
  os: string;
  device_type: string;
  created_at: string;
};

export type VisitorResponse = {
  code: number;
  data: VisitorItem[];
};

export type VisitorCountResponse = {
  code: number;
  count: number;
};

/** 获取最近访客列表 */
export const getVisitors = (params?: { page?: number; size?: number }) => {
  return http.request<VisitorResponse>("get", "/api/visitors", { params });
};

/** 获取总访客数 */
export const getVisitorCount = () => {
  return http.request<VisitorCountResponse>("get", "/api/visitors/count");
};

/** 删除单条访客记录 */
export const deleteVisitor = (visitorId: number) => {
  return http.request<{ code: number }>("delete", `/api/visitors/${visitorId}`);
};

/** 清空所有访客记录 */
export const clearVisitors = () => {
  return http.request<{ code: number }>("delete", "/api/visitors");
};

File: Kirameku-backend/app/models/visitor.py
from datetime import datetime
from typing import Optional
from sqlmodel import SQLModel, Field


class Visitor(SQLModel, table=True):
    __tablename__ = "visitor"

    id: Optional[int] = Field(default=None, primary_key=True)
    ip: str = Field(max_length=45)
    path: str = Field(default="", max_length=500)
    user_agent: str = Field(default="")
    city: str = Field(default="", max_length=100)
    region: str = Field(default="", max_length=100)
    country: str = Field(default="", max_length=100)
    district: str = Field(default="", max_length=100)
    org: str = Field(default="", max_length=200)
    asn: str = Field(default="", max_length=50)
    is_mobile: bool = Field(default=False)
    is_proxy: bool = Field(default=False)
    is_hosting: bool = Field(default=False)
    browser: str = Field(default="", max_length=50)
    os: str = Field(default="", max_length=50)
    device_type: str = Field(default="", max_length=20)
    created_at: datetime = Field(default_factory=datetime.now)
